### ISSUE TITLE
Narrator doesn't announce a text mouse selection in DataGridVIewTextBoxCell and ToolStripTextBox

### ISSUE THREAD

**Issue by vladimir-krestov** [dotnet/winforms#2794]
* .NET Core Version: 5.0.100-alpha.1.20073.10

* Have you experienced this same bug with .NET Framework?: TextPattern hasn't implemented in .Net 4.8 yet.

**Problem description:**
- Narrator doesn't read symbols and a selected text range in `DataGridViewTextBoxCell` and `ToolStripTextBox` edit controls when using a keyboard or mouse selecting
This seems to be due to incorrect handling of keyboard arrow keys pressing.
(Please check `ProcessDialogKey` method)
But if you change the focus and return then Narrator chooses the correct selected range. 
![ToolStrip](https://user-images.githubusercontent.com/49272759/73546925-198d0980-444f-11ea-8a32-287d269e9044.gif)

**Expected behavior:**
- Narrator reads symbols and a selected text range as in `TextBox`, `ComboBox`
![TextBox](https://user-images.githubusercontent.com/49272759/73547850-cf0c8c80-4450-11ea-8428-01be040e45dc.gif)

**Minimal repro:**
- Create a .Net Core 5.0 app
- Add ToolStrip or DataGridView with TextBox items
- Run app and Narrator
- Check text announcing

**Comment by vladimir-krestov** [comment]
Related PR #2701

**Comment by merriemcgaw** [comment]
In general Narrator users won't be using a mouse to navigate the DGV. Let's close this one until/unless we get specific feedback from a customer or compliance tester that we need to do something here. This is a high risk, high cost issue so we will do it if/when it is raised by customer.
